cmd/object-handlers.go

```diff
@@ -2923,7 +2939,11 @@ func (api objectAPIHandlers) PutObjectLegalHoldHandler(w http.ResponseWriter, r
 		return
 	}
 	if replicate {
-		globalReplicationState.queueReplicaTask(objInfo)
+		if sync {
```

Comment:
This pattern can be changed, we can have a func `scheduleReplication` which will block for sync or queue and return for async.

Reply:
done